File: csharp/BufferHashBenchmark.cs
public abstract class BufferHashBenchmark : Benchmark
{
    protected byte[] _data;
    protected int _n;
    protected uint _result;

    protected BufferHashBenchmark()
    {
        _data = Array.Empty<byte>();
        _result = 0;
        _n = (int)ConfigVal("size");
    }

    public override void Prepare()
    {
        _data = new byte[_n];
        for (int i = 0; i < _data.Length; i++)
        {
            _data[i] = (byte)Helper.NextInt(256);
        }
    }

    protected abstract uint Test();

    public override void Run(long IterationId)
    {
        _result += Test();
    }

    public override uint Checksum => _result;
    public override string TypeName => "Hash";
}

File: swift/Sources/benchmarks/BufferHashBenchmark.swift
import Foundation

class BufferHashBenchmark: BenchmarkProtocol {
  var data: [UInt8] = []
  private var sizeVal: Int64 = 0
  private var resultVal: UInt32 = 0

  init() {

  }

  func prepare() {
    if sizeVal == 0 {
      sizeVal = configValue("size") ?? 0
      data.reserveCapacity(Int(sizeVal))
      for _ in 0..<Int(sizeVal) {
        data.append(UInt8(Helper.nextInt(max: 256)))
      }
    }
  }

  func test() -> UInt32 {
    return 0
  }

  func run(iterationId: Int) {
    resultVal &+= test()
  }

  var checksum: UInt32 {
    return resultVal
  }

  var name: String { return "BufferHashBenchmark" }
}
